File: Optional(lab5)/src/main/java/com/epam/poker/scala/Main.scala
package com.epam.poker.scala

object Main {

  def main(args: Array[String]): Unit = {

    var playersBids: List[Option[Bid]] = List.tabulate(10)(n => if (n % 2 == 1) Some(Bid("Player Alex" + n, n * 500)) else None)
    var observersBid: List[Option[Bid]] = List.tabulate(10)(n => if (n % 2 == 1) Some(Bid("Observer John" + n, n * 1000)) else None)
    var emptyBids: List[Option[Bid]] = List(None, None)

    println(PokerService.getWinner(playersBids, observersBid))
    println(PokerService.getWinner(emptyBids, observersBid))
    println(PokerService.getWinner(emptyBids, emptyBids))

  }

}

File: Optional(lab5)/src/main/java/com/epam/poker/java/Main.java
package com.epam.poker.java;

import java.util.List;
import java.util.Optional;

public class Main {
    public static void main(String[] args) {

        List<Optional<Bid>> playerBids = List.of(
                Optional.of(Bid.builder().amount(300).bidder("Alex").build()),
                Optional.of(Bid.builder().amount(1200).bidder("Alex1").build()),
                Optional.empty(),
                Optional.of(Bid.builder().bidder("Alex3").build()),
                Optional.of(Bid.builder().amount(600).bidder("Alex4").build()),
                Optional.empty(),
                Optional.of(Bid.builder().bidder("Alex6").build())
        );

        List<Optional<Bid>> observerBids = List.of(
                Optional.of(Bid.builder().amount(300).bidder("John").build()),
                Optional.of(Bid.builder().amount(1200).bidder("John1").build()),
                Optional.empty(),
                Optional.empty()
        );

        List<Optional<Bid>> emptyOptionals = List.of(
                Optional.empty(),
                Optional.empty()
        );


        PokerService.getWinner(playerBids, observerBids)
                .ifPresentOrElse(
                        bidder -> System.out.println("The bidder " + bidder.getBidder() + " Won!"),
                        () -> System.out.println("The was no bidders")
                );

        PokerService.getWinner(emptyOptionals, observerBids)
                .ifPresentOrElse(
                        bidder -> System.out.println("The bidder " + bidder.getBidder() + " Won!"),
                        () -> System.out.println("The was no bidders")
                );

        PokerService.getWinner(emptyOptionals, emptyOptionals)
                .ifPresentOrElse(
                        bidder -> System.out.println("The bidder " + bidder.getBidder() + " Won!"),
                        () -> System.out.println("The was no bidders")
                );

    }
}
